src/ducks/Studio/url.js

```diff
@@ -1,14 +1,24 @@
 import { stringify, parse } from 'query-string'
 import { DEFAULT_SETTINGS, DEFAULT_OPTIONS } from './defaults'
-import { Events, Metrics, compatabilityMap } from '../SANCharts/data'
+import { Events, Metrics, compatabilityMap } from '../SANCharts/metrics/data'
 import { buildCompareKey } from './Compare/utils'
+import { Submetrics } from '../SANCharts/metrics/submetrics'
 
 const { trendPositionHistory } = Events
 
 const COMPARE_CONNECTOR = '-CC-'
 const getMetricsKeys = metrics => metrics.map(({ key }) => key)
 const toArray = keys => (typeof keys === 'string' ? [keys] : keys)
-const convertKeyToMetric = (key, dict) => dict[key] || compatabilityMap[key]
+const convertKeyToMetric = (key, dict) =>
+  dict[key] || compatabilityMap[key] || searchFromSubmetrics(key)
+
+function searchFromSubmetrics (key) {
+  for (let metricKey of Object.keys(Submetrics)) {
+    let list = Submetrics[metricKey]
+    const found = list.find(({ key: subMetricKey }) => subMetricKey === key)
+    if (found) return found
+  }
+}
```

Comment:
List of submetrics could be accessed directly without additional lookup by iterating over object values with  ` for (const submetrics of Object.values(Submetrics))` 

```js
function searchFromSubmetrics(metricKey) {
  for (const submetrics of Object.values(Submetrics)) {
    const found = submetrics.find(({ key }) => key === metricKey)

    if (found) return found
  }
}
```

Reply:
@DmitriVanGuard , do't understand. Please, explain?